portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -29,16 +35,22 @@
 
   private ArrayList<Comment> comments;
   
-  @Override
-  public void init() {
-    comments = new ArrayList<Comment>();
-  }
-  
   /* responds with a JSON string containing authors and comments*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Convert the ArrayLists to JSON
+    // Query datastore for all comments
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // sort timestamps such that oldest comments are displayed first
+    Query commentQuery = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery storedComments = datastore.prepare(commentQuery);
+    comments = new ArrayList<Comment>();
+
+    for (Entity entity : storedComments.asIterable()) {
+        Comment comment = new Comment((String) entity.getProperty("author"), (String) entity.getProperty("value"));
+        comments.add(comment);
+    }    
```

Comment:
There could be a method that takes a PreparedQuery and return the list of comments.

Reply:
Good idea, added as a method to the new CommentDatabase class.